src/components/WatchlistCopy/WatchlistCopy.js

```diff
@@ -0,0 +1,125 @@
+import React, { PureComponent } from 'react'
+import WatchlistCopyBtn from './WatchlistCopyBtn'
+import WatchlistCopyPopup from './WatchlistCopyPopup'
+import { connect } from 'react-redux'
+import { Popup } from 'semantic-ui-react'
+import * as actions from '../../actions/types'
+
+const style = {
+  padding: 0
+}
+
+class WatchlistCopy extends PureComponent {
+  state = {
+    isPopupVisible: false,
+    newWatchlistTitle: '',
+    assetsToCopy: new Set()
+  }
+
+  static getDerivedStateFromProps (nextProps, prevState) {
+    return {
+      ...prevState,
+      assetsToCopy: new Set(nextProps.assets.map(({ id }) => id))
+    }
+  }
+
+  onWatchlistTitleChange = ({ currentTarget }) => {
+    this.setState(prevState => ({
+      ...prevState,
+      newWatchlistTitle: currentTarget.value
+    }))
+  }
+
+  onAssetClick = id => {
+    const assetsToCopy = new Set([...this.state.assetsToCopy])
+
+    if (assetsToCopy.has(id)) {
+      assetsToCopy.delete(id)
+    } else {
+      assetsToCopy.add(id)
+    }
+
+    this.setState(prevState => ({
+      ...prevState,
+      assetsToCopy
+    }))
+  }
+
+  handleCopyConfirm = evt => {
```

Comment:
You used handle* and on* word for handlers func in the single class. @DmitriVanGuard